### PR TITLE
Deeplinks using browser location.hash

### PR DESCRIPTION
**Pull request by bunnymatic**
Move best_of show handling onto the root page

    * use location.hash to pass the id of the 'best of' that you want to see
    * clear the hash when a new cb is generated
    * added best_of_url helper method (not in routes) that builds anchored link on root (/#<cb_id>)
    * add hash watcher in js on page load.
    * remove clickbaits#show
    * move javascript functions into a namespace (so they could be tested more easily)

### CODE REVIEW

**Comment by shannonwells** [comment]
IIRC the upshot of our discussion this morning (2/14) was, we can keep/put back best_of route just for a ranked, sortable listing of headlines, and we don't need the show page at all, and you should keep going the route you're going.

**Comment by bunnymatic** [comment]
me thinks this is ready to roll

File: app/assets/javascripts/home.js [hunk 1]
```diff
@@ -56,4 +61,11 @@ window.onload = function () {
         count: 100,
         overlap: 20
     });
+
+    // fetch best_of if the anchor tag is present
+    if (location.hash) {
+        CBG.getBestOf(location.hash.substr(1));
```
Comment (shannonwells) [inline app/assets/javascripts/home.js]: 👍

**Comment by shannonwells** [comment]
bitmoji

**Pull request merged**